rmw_zenoh_cpp/src/detail/graph_cache.cpp

```diff
@@ -1208,7 +1208,26 @@ rmw_ret_t GraphCache::get_entities_info_by_topic(
           }
         }
 
-        // TODO(Yadunund): Set gid.
+        liveliness::ConstEntityPtr entity;
+        if (entity_type == EntityType::Publisher) {
+          entity = (*topic_data->pubs_.begin());
+        } else {
+          entity = (*topic_data->subs_.begin());
+        }
+
+        if (entity) {
+          uint8_t gid[RMW_GID_STORAGE_SIZE];
+          entity->get_gid(gid);
```

Comment:
Should we iterate though all the `Entity` ptrs in the `EnittySet`?
```
const auto & entity_set = entity_type == EntityType::Publisher? topic_data->pubs_ : topic_data->subs_;
for (auto entity_it = entity_set.begin(); entity_it != entity_set.end(); ++entity_it) {
   uint8_t gid[RMW_GID_STORAGE_SIZE];
   entity->get_gid(gid);
    ret = rmw_topic_endpoint_info_set_gid(
        &endpoint_info,
        gid,
        RMW_GID_STORAGE_SIZE
     );
     if (RMW_RET_OK != ret) {
        return ret;
          }
}

```

Reply:
If we iterate over the list, then it will override the gid value (if there is more than one).